core/src/main/java/pl/agh/edu/utils/CustomBigDecimal.java

```diff
@@ -0,0 +1,124 @@
+package pl.agh.edu.utils;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.math.RoundingMode;
+import java.util.Arrays;
+import java.util.List;
+
+import ch.obermuhlner.math.big.BigDecimalMath;
+
+public class CustomBigDecimal {
+	private BigDecimal value;
+	private final MathContext mc = new MathContext(100, RoundingMode.HALF_UP);
+
+	public CustomBigDecimal(String val) {
+		this.value = new BigDecimal(val);
+	}
+
+	public CustomBigDecimal(BigDecimal bigDecimal) {
+		this.value = bigDecimal;
+	}
+
+	public CustomBigDecimal(float val) {
+		this.value = new BigDecimal(val, mc);
+	}
+
+	public CustomBigDecimal add(CustomBigDecimal other) {
+		return new CustomBigDecimal(this.value.add(other.value));
+	}
+
+	public CustomBigDecimal subtract(CustomBigDecimal other) {
+		return new CustomBigDecimal(this.value.subtract(other.value));
+	}
+
+	public CustomBigDecimal multiply(CustomBigDecimal other) {
+		return new CustomBigDecimal(this.value.multiply(other.value));
+	}
+
+	public CustomBigDecimal divide(CustomBigDecimal other) {
+		if (other.value.equals(BigDecimal.ZERO)) {
+			throw new ArithmeticException("Division by zero");
+		}
+		return new CustomBigDecimal(this.value.divide(other.value, mc));
+	}
+
+	public int compareTo(CustomBigDecimal other) {
+		return this.value.compareTo(other.value);
+	}
+
+	public CustomBigDecimal pow(BigDecimal n) {
+		return new CustomBigDecimal(BigDecimalMath.pow(value, n, mc));
+	}
+
+	public CustomBigDecimal log10() {
+		return new CustomBigDecimal(BigDecimalMath.log10(this.value, mc));
+	}
+
+	@Override
+	public String toString() { // 123 456 = 123k
+		int zeros = BigDecimalMath.log10(this.value, mc).intValue();
+		int triZeros = zeros / 3;
+		Prefix prefix = Prefix.getPrefixByValue(BigDecimalMath.pow(BigDecimal.valueOf(1000), BigDecimal.valueOf(triZeros), mc));
+		String number = value.toString().split("\\.")[0];
+		int tmp = prefix.getValue().toString().length();
+		String beforeComa = number.substring(0, number.length() - tmp + 1);
+		String result = beforeComa;
+		if (prefix.getValue().compareTo(BigDecimal.ONE) > 0) {
+			String afterComa = number.substring(number.length() - tmp + 1, number.length());
+			result += beforeComa.length() >= 3 ? "" : ("." + afterComa.substring(0, 3 - beforeComa.length()));
+			result += prefix.name();
+		}
+		return result;
+	}
+
+	public CustomBigDecimal roundToStringValue() { // 123 456 = 123 000 = 123 k
```

Comment:
jeśli dobrze rozumiem to ta metoda służy do tego żeby ustawić bigDecimala tak zeby był równy reprezentacji graficznej czyli: 123456 -> 123000 Niemożna zatem zrobić po prostu value = value / prefix.getValue() * prefix.getValue() ?

Reply:
nie do końca to będzie działać bo chcemy mieć tam w niektórych przypadkach 1 lub 2 cyfry po przecinku. Ale postaram się uprościć to co jest